### PR TITLE
Refactor parseNextLayer to include parseUntilLayer and parseUntil

### PR DESCRIPTION
**Pull request by rtmeng**
This pull request implements the fix to #657 by refactoring the function `parseNextLayer()` to include parameters `parseUntil` and `parseUntilLayer`, which fixes the issue of `Packet::setRawPacket` parsing layers it should not when `parseUntilLayer` is set. Like mentioned in the Issue, this solution doesn't break the API and relies on the layers to not parse too far if it's not requested. 

Main Changes:
- Changed each layer's `parseNextLayer()` to check if the next layer should be parsed
- Fixed inconsistencies of using the keyword override

### CODE REVIEW

File: Packet++/src/BgpLayer.cpp [hunk 1]
```diff
@@ -76,12 +76,15 @@ std::string BgpLayer::getMessageTypeAsString() const
...
 
-void BgpLayer::parseNextLayer()
+void BgpLayer::parseNextLayer(ProtocolType parseUntil, OsiModelLayer parseUntilLayer)
 {
 	size_t headerLen = getHeaderLen();
 	if (m_DataLen <= headerLen || headerLen == 0)
 		return;
 
+	if (getProtocol() == parseUntil || getOsiModelLayer() == parseUntilLayer)
+        return;
```
Comment (egecetin) [inline Packet++/src/BgpLayer.cpp:86]: It will be a pain to change a bit but all `return` statements in cpp files unfortunately requires a tab indentation

Comment (rtmeng) [reply Packet++/src/BgpLayer.cpp:86]: Got it, somehow didn’t notice that locally, I’ll go through and add them

**Comment by tigercosmos** [comment]
Could you also add tests for these changes?

**Comment by seladb** [comment]
@rtmeng do you plan to continue working on this PR?

**Comment by rtmeng** [comment]
Sorry just saw this @seladb, I won’t be able to in the next month so I can close this PR and reopen it later if still applicable.

**Pull request closed without merging**